### PR TITLE
HPCC-24975 Improve ambiguous '--name' parameters on ecl command line

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/scm/ws_workunits_queryset_req_resp.ecm [hunk 1]
```diff
@@ -101,12 +101,15 @@ ESPrequest [nil_remove] WUPublishWorkunitRequest
...
     string Cluster;
+    [min_ver("1.81")] string QueryName; //if set always used for the query name
+    [min_ver("1.81")] string WorkUnitJobName;  //if set always used for the workunit job name field
     string JobName;
     int Activate;
     bool NotifyCluster(false);
     int Wait(10000);
     bool NoReload(0);
-    bool UpdateWorkUnitName(0);
+    bool UpdateWorkUnitName(0);     //Update Workunit JobName field from JobName if WorkUnitJobName is blank (for backward comppatability)
```
Comment (wangkx) [inline esp/scm/ws_workunits_queryset_req_resp.ecm:111]: compatibility

File: ecl/eclcmd/eclcmd_core.cpp [hunk 2]
```diff
@@ -448,8 +453,15 @@ class EclCmdPublish : public EclCmdWithEclTarget
...
 
-        if (optName.length())
-            req->setJobName(optName.get());
+        if (optJobName.length())
+            req->setWorkUnitJobName(optJobName);
+        if (optQueryName.length())
+        {
+            //very confusing parameter name, but set for backward compatability
+            //  for old servers... will not override WorkUnit JobName from call to doDeploy above, so that job name will still win
+            req->setJobName(optQueryName);
```
Comment (wangkx) [inline ecl/eclcmd/eclcmd_core.cpp:462]: Is this for old server?

**Review by wangkx — COMMENTED** [review]
one question and one minor

**Comment by richardkchapman** [comment]
@afishbeck Can you comment?

**Comment by richardkchapman** [comment]
@afishbeck seems to be stalled. Should this target master?

**Comment by afishbeck** [comment]
I'll switch target to master and refresh my memory so I can respond to Kevin's question.

File: ecl/eclcmd/eclcmd_core.cpp (hunk 2 above)
Comment (afishbeck) [reply ecl/eclcmd/eclcmd_core.cpp:462]: Yes, setting this keeps old servers working as intended.

**Comment by afishbeck** [comment]
@richardkchapman fixed typo

**Review by wangkx — APPROVED** [review]
@afishbeck looks fine

**Pull request merged**